ports/webassembly/library.js

```diff
@@ -33,10 +33,9 @@ mergeInto(LibraryManager.library, {
                 process.stdout.write(b);
             } else {
                 var c = String.fromCharCode(getValue(ptr + i, 'i8'));
-                var mp_js_stdout = document.getElementById('mp_js_stdout');
-                var print = new Event('print');
-                print.data = c;
-                mp_js_stdout.dispatchEvent(print);
+                var printEvent = new Event('micropython-print');
+                printEvent.data = c;
+                document.dispatchEvent(printEvent);
```

Comment:
Should we change the check for `typeof window === 'undefined'` above to check document instead?

Do we also want to support web workers that don't have a document or window object?

Reply:
Yeah... so running MicroPython in web-workers is the next blip on my radar. ;-) There are ways to detect if something is running inside a web worker (in which case we should `postMessage(foo)` instead?). I guess yet another branch in this conditional.

Reply:
I don't think we should assume that stdout should be tied to `postMessage` in a web worker. Someone might want to keep the stdout in in the worker thread.

Thinking about it more, using an event probably makes sense in nodejs as well so that stdout from MicroPython could be redirected somewhere else besides stdout of nodejs.

So maybe no branches are needed?

Reply:
It might also be more efficient to collect all characters into one event, rather than having a single event per character.  And better to make that change now, along with this event change, since all these changes are backwards incompatible.

So the idea would be:
1. remove `mp_js_stdout` DOM element
2. remove special case for node, just always emit an event (then when running node it hooks the event to forward the data to `process.stdout`)
3. put all characters in the one event

I know that's creeping the scope of this PR, but I think it's the right direction to move in.  Let me know what you think @ntoll.

Reply:
I agree this is the right direction.

1. I done in this PR.
2. I think we should proceed with care and wrap this in another PR. There is yet another sort of behaviour to account for: when MicroPython is run within a web worker context (i.e. we're in a browser, but there's no `browser` object to use for dispatching events).
3. This makes sense, but should probably be done in a separate PR just to keep things tidy/avoid scope creep.

Reply:
I forgot to mention I'm currently at a conference/travelling, but can look at 2 and 3 next week. (Greetings from Sweden :sweden: )